controller/Review.php

```diff
@@ -103,7 +103,11 @@ public function index(): void
         if ($this->isSubmissionReviewRequestMessageProvided()) {
             $deliveryId = $this->getDeliveryId();
             $userId = $this->getUserId();
-            $execution = $finder->findLastExecutionByUserAndDelivery($userId, $deliveryId);
+            $resourceLinkId = null;
+            if ($launchData->hasVariable(LtiLaunchData::RESOURCE_LINK_ID)) {
```

Comment:
This condition and the assignment below look a little bit counter-intuitive. I suppose it's done this way, 'cause the session always has some resource link ID generated, even if there's none provided in the parameters, right?

Reply:
nope, as i understand in case of `SubmissionReviewRequest` no `resourceLinkId` was generated and in this case 
```PHP
$this->ltiSession->getLtiLinkResource();
```
will throw exception. also this service will return `Resource` entity which required for service which used for fetching delivery executions.
we can make this place exception driven.

Reply:
I don't mind leaving it as-is.